TombLib/TombLib/Catalogs/TEN Node Catalogs/Revolution.lua

```diff
@@ -0,0 +1,180 @@
+--Revolution Nodes by TrainWreck; Code references by Lwmte, Adngel. Shoutout to DaviDMRR, RichardTRLE and Stranger1992 for the help.
+
+local Timer = require("Engine.Timer")
+
+-- Construct timed transform data and start transform
+LevelFuncs.Engine.Node.ConstructTimedData = function(objectName, objectcentre, dataType, radius, time, startangle, endangle, isLoop, isCCW, isSmooth, isRotate, isCRotate, SFX)
+
+    local dataName  = objectName .. "_revolve_data"
+    
+    	if LevelVars[dataName] and LevelVars[dataName].Timer then
+		if LevelVars[dataName].Timer:IsActive() then
+        	return
+		end
+		Timer.Delete(LevelVars[dataName].Name)
+		LevelVars[dataName] = nil
+	end
+
+	LevelVars[dataName] = {}
+    LevelVars[dataName].Progress   = 0
+    LevelVars[dataName].Interval   = 1 / (time * 30)
+    LevelVars[dataName].DataType   = dataType
+    LevelVars[dataName].Radius     = radius
+    LevelVars[dataName].ObjectName = objectName
+    LevelVars[dataName].CentreName = objectcentre
+    LevelVars[dataName].Name       = dataName
+    LevelVars[dataName].Timer      = Timer.Create(dataName, 1 / 30, true, false, LevelFuncs.Engine.Node.TransformTimedData, dataName)
+    LevelVars[dataName].StartAngle = startangle
+    LevelVars[dataName].EndAngle   = endangle
+    LevelVars[dataName].CCW        = isCCW

```

Comment:
Its not obvious what CCW is. Personally rename this symbol to `counterClockwise`

Reply:
The explanation for the variable is in the node. Also added comments at variable declaration.